### DISCUSSION TITLE
S3 storage type

### DISCUSSION THREAD

**Discussion by craftzneko** [outline/outline#5678] [Self Hosting] [answered]
does anybody know if outline uses presigned URLs or post policy for its uploads to S3 storage. I am trying to find proxy to send requests to azure and knowing the difference would be helpful. I see in the POST request that there is a header that seems to suggest there is a post policy but the type script code here mentioned using presigned urls.

**Comment by tommoor** [comment, marked as answer]
It's both, a presigned post policy. I'm not sure it's possible to have a presigned upload url without using a post policy

**Comment by craftzneko** [comment]
Thanks @tommoor I assume this is the type of request that outline is making to s3 ?
https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
